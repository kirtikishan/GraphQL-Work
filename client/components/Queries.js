import gql from 'graphql-tag';


export const homePageQuery = gql`
  {
     carOfTheWeek{
          modelId,
          review,
          carModel {
            makeId,
            name,
            price,
            imageUrl
            make {
              name
            }
          }
     }
  }
`;

export const searchQuery = gql`
      query search {
         makes {
            id,
            name
            models {
                id,
                name
            }
         }
      }
    `;


export const detailsQuery = gql`
  query ModelsQuery($id : Int!) {
    model(id: $id) {
      name,
      price,
      imageUrl,
      make {
          id,
          name
      }
    }
  }
`;