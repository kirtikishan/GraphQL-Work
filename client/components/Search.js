import React, {Component} from 'react';

class Search extends Component {
    render() {
        return (
            <div>Search</div>
        )
    }
}

const query = gql`
  {
     carOfTheWeek {
        modelId,
        review,
        carModel {
            makeId,
            name
        }
     }     
  }
`;

export default Search;