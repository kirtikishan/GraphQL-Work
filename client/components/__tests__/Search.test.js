import * as React from 'react';
import {Search} from '../Search';
import { shallow } from 'enzyme';
import {searchQuery} from '../Queries';

describe('Search Component', () => {
    it('should render the search correctly', () => {
        const data = {
            data: {
                makes: [
                    {
                        id: 10,
                        name: "Porsche",
                        models: [
                            {
                                id: 100,
                                name: "911 Carrera 4s"
                            },
                            {
                                id: 100,
                                name: "911 Carrera 4s"
                            },
                            {
                                id: 100,
                                name: "911 Carrera 4s"
                            }
                        ]
                    }
                ]
            }
        };

        const wrapper = shallow(
            <Search data={data}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should be rendered the correct search query', () => {
        expect(searchQuery).toMatchSnapshot();
    });
})
;
