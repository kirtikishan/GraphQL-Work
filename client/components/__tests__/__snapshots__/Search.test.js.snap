// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should be rendered the correct search query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "search",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "makes",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "name",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "models",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "name",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 171,
    "start": 0,
  },
}
`;

exports[`Search Component should render the search correctly 1`] = `
<div
  className="search-page"
>
  <div
    className="search-page__head-text"
  >
    Our Range
  </div>
  <div
    className="search-page__head-text2"
  >
    Please select a car for details
  </div>
  <div
    className="search-page__selection-container"
  >
    <div
      className="search-page__selection-box1"
    >
      <span
        className="search-page__make-text"
      >
        Please select car make
      </span>
      <Select
        addLabelText="Add \\"{label}\\"?"
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="make-select"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={false}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="makes"
        noResultsText="No results found"
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={Array []}
        pageSize={5}
        placeholder="Select..."
        required={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        value="Select"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <div
      className="search-page__selection-box2"
    >
      <span
        className="search-page__model-text"
      >
        Please select car model
      </span>
      <Select
        addLabelText="Add \\"{label}\\"?"
        arrowRenderer={[Function]}
        autosize={true}
        backspaceRemoves={true}
        backspaceToRemoveMessage="Press backspace to remove {label}"
        className="model-select"
        clearAllText="Clear all"
        clearRenderer={[Function]}
        clearValueText="Clear value"
        clearable={false}
        closeOnSelect={true}
        deleteRemoves={true}
        delimiter=","
        disabled={true}
        escapeClearsValue={true}
        filterOptions={[Function]}
        ignoreAccents={true}
        ignoreCase={true}
        inputProps={Object {}}
        isLoading={false}
        joinValues={false}
        labelKey="label"
        matchPos="any"
        matchProp="any"
        menuBuffer={0}
        menuRenderer={[Function]}
        multi={false}
        name="model"
        noResultsText="No results found"
        onBlurResetsInput={true}
        onChange={[Function]}
        onCloseResetsInput={true}
        onSelectResetsInput={true}
        openOnClick={true}
        optionComponent={[Function]}
        options={Array []}
        pageSize={5}
        placeholder="Select..."
        required={false}
        scrollMenuIntoView={true}
        searchable={false}
        simpleValue={false}
        tabSelectsValue={true}
        value="Select"
        valueComponent={[Function]}
        valueKey="value"
      />
    </div>
    <span
      className="search-page__detail-btn disabled"
    >
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/make/model/"
      >
        View Detail
      </Link>
    </span>
  </div>
</div>
`;
